### PR TITLE
GODRIVER-2978 Automate syncing code between Go driver v1 and v2

### PR DESCRIPTION
**Pull request by blink1073**
GODRIVER-2978

## Summary
Help automate syncing between the v1 branch and master, which is targeting v1.

## Background & Motivation
Help make the v2 transition smoother.

### CODE REVIEW

File: etc/cherry-picker.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+sha=$1
+base=v1
+target=master
+dirname="/tmp/go-driver-$(openssl rand -hex 12)"
```
Comment (matthewdale) [inline etc/cherry-picker.sh:6]: Optional: Consider using `mktemp -d` instead, which honors the temp dir on different OSes.

**Review by matthewdale — APPROVED** [review]
Looks good 👍 

Alternatively, could an existing action like https://github.com/marketplace/actions/sync-branches work as well?

File: etc/cherry-picker.sh (hunk 1 above)
Comment (blink1073) [reply etc/cherry-picker.sh]: Done

**Comment by blink1073** [comment]
Updated to use our own action, since [github.com/marketplace/actions/sync-branches](https://github.com/marketplace/actions/sync-branches) is unmaintained, and this allows us to use the same script locally when needed.

**Review by matthewdale — APPROVED** [review]
Looks good 👍

**Pull request merged**